include/silo/query_engine/filter_expressions/and.h

```diff
@@ -0,0 +1,27 @@
+#ifndef SILO_AND_H
+#define SILO_AND_H
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "silo/query_engine/filter_expressions/expression.h"
+
+namespace silo::query_engine::filter_expressions {
+
+struct And : public Expression {
+   std::vector<std::unique_ptr<Expression>> children;
+
+   explicit And(std::vector<std::unique_ptr<Expression>>&& children);
```

Comment:
Because the vector contains unique_ptrs the compiler does not let us pass a reference to the vector, as we would then have multiple references to the unique_ptrs inside. Therefore we can only accept a temporary object (the caller of the constructor gives up ownership of the vector) that contains the unique_ptrs

Reply:
Only a short follow up question: Why do we need unique_ptrs here anyway. Cant we use shared_ptr here? The overhead from both should be the same.